clients/ts/signalr-protocol-msgpack/src/MessagePackHubProtocol.ts

```diff
@@ -37,7 +37,19 @@ export class MessagePackHubProtocol implements IHubProtocol {
         if (logger === null) {
             logger = NullLogger.instance;
         }
-        return BinaryMessageFormat.parse(input).map((m) => this.parseMessage(m, logger));
+
+        const messages = BinaryMessageFormat.parse(input);
+
+        const hubMessages = [];
+        for (const message of messages) {
+            const parsedMessage = this.parseMessage(message, logger);
+            // Can be null for an unknown message
```

Comment:
Add note that `parseMessage` already logged about this. Was about to ask for you to log this condition here ;)